dashboard/app/controllers/activities_controller.rb

```diff
@@ -31,7 +31,9 @@ def milestone
     # disabled. (A cached view might post to this action even if milestone posts
     # are disabled in the gatekeeper.)
     enabled = Gatekeeper.allows('postMilestone', where: {script_name: script_name}, default: true)
-    unless enabled
+    # Exception: Always post milestone if we passed the last level of the script.
+    # Keep this logic in sync with code-studio/reporting#sendReport on the client.
+    unless enabled || (solved && @script_level.try(:final_level?))
```

Comment:
Feels a little weird to me to have a gatekeeper setting that can disable milestone posts, but then still have it let some through. Will we ever be in a situation where we would like to disable ALL milestone posts (including on final levels)?

If so, I wonder if we'd be better off having two settings.

Otherwise changes lgtm